specification/synonyms/_types/SynonymRule.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Id } from '@_types/common'
+
+export type SynonymString = string
+
+export class SynonymRuleOptionalId {
```

Comment:
There are operations (PUT) that don't need an ID to be included, so it will be generated by the API. I wanted to use a strict type for that, but it might be overkill...

Reply:
My two cents: I don't think it hurts, but I'm not sure what it adds. Happy to leave it in but I don't have strong feelings either way.

Reply:
Thanks @Anaethelion - I added comments to clarify why they're used 👍